Automattic/jetpack
# Simplify error notices for broken connections

#16194 introduced the ability for Jetpack to identify, confirm and handle XML-RPC errors. It also introduced admin notices to display such errors to users. 

Up until now we were only displaying errors for Blog token issues. This PR adds error messages for user tokens errors too.

### Refactor

This PR also simplifies the code that was displaying the errors. At first, there was this idea of having an abstraction that would allow us to treat each error code differently, having a class to handle each one of them.

However, this abstraction was in place to handle theoretical situations while, in practice, we were simply displaying the same generic error message in all case. Therefore I decided to get rid of the abstraction for now and keep it simple. When needed, it's easy to add a per error code handling there.

**Note:** I think it's completely safe to just remove the "blog token error handling" class and all the hooks introduced in 8.7 because they have not been used anywhere or by any other plugin using the connection package.

#### Changes proposed in this Pull Request:

* Adds user feedback for XML-RPC user token related errors

#### Jetpack product discussion


p1HpG7-9q9-p2

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Start with a connected Jetpack site and the Jetpack Debug helper plugin activated
* Go to Broken Token and click "Set invalid user token (current user)"
* Go to XML-RPC Errors.
* Click the link to visit the Jetpack debugger
* Back to XML-RPC error, confirm you have an error under "Verified error"
* Visit any Dashboard page and confirm you see the error in admin notices
* Visit the Jetpack Dashboard and confirm you see the error at the top of the page along with a CTA to restore the connection

#### Proposed changelog entry for your changes:

* Adds user feedback for XML-RPC user token related errors
